app/containers/EditStudentContainer.js

```diff
@@ -0,0 +1,126 @@
+import React, {Component} from 'react';
+import store from '../store';
+import EditStudent from '../components/EditStudent';
+import axios from 'axios';
+import {
+  receiveStudents,
+  updateStudent,
+  setFirstStudentName,
+  setLastStudentName,
+  setStudentEmail,
+  setStudentCampus,
+  getStudentById,
+} from '../action-creators/students';
+import {
+  receiveCampuses,
+} from '../action-creators/campuses';
+
+const initialState = {
+  studentFirstName: '',
+  studentLastName: '',
+  studentEmail: '',
+  studentCampus: '',
+};
+
+class EditStudentContainer extends Component {
+  constructor() {
+    super();
+    this.state = store.getState();
+    this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
+    this.handleLastNameChange = this.handleLastNameChange.bind(this);
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+    this.handleCampusChange = this.handleCampusChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribe = store.subscribe(() =>
+      this.setState(store.getState())
+    )
+    Promise.all([
+      axios.get('/api/campuses'),
+      axios.get('/api/students')
+    ])
+      .then(responses => responses.map(res => res.data))
+      .then(([campuses, students]) => {
+        store.dispatch(receiveCampuses(campuses));
+        store.dispatch(receiveStudents(students));
+      });
+      store.dispatch(getStudentById(this.props.match.params.id))
+      // console.log('this.props.match.params.id: ', this.props.match.params.id);
+    // this.setState({studentFirstName: store.getState().students.selectedStudent.lastName});
+    // this.setState({studentEmail: store.getState().students.selectedStudent.email});
+    // this.setState({studentCampus: store.getState().students.selectedStudent.campus && store.getState().students.selectedStudent.campus.name});
+  }
+
+  componentWillUnmount() {
+     this.unsubscribe();
+  }
+
+  handleFirstNameChange(event) {
+    // this.setState({name:event.target.value});
+     console.log('handleNameChange - event: ', event.target)
+    const val = event.target.value;
+    store.dispatch(setFirstStudentName(val));
+    this.setState({studentFirstName: store.getState().students.firstName});
```

Comment:
We don't want to be calling `setState` right after a `dispatch`. The state for this component should come from the store. Dispatching the event should have the effect of changing the state in the store, which would then flow through to the state of the component because the component has subscribed to the store.